The terraform-provider-azurerm_v2.73.0_x5.exe plugin crashed

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version
* Terraform v1.0.6
* AzureRM v2.75.0
<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_container_registry`
 * `azurerm_frontdoor`
 * `azurerm_storage_account`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
storage_account:
resource "azurerm_storage_account" "storage_account" {
  name                     = var.storage_account_name
  resource_group_name      = var.rg_name
  location                 = var.location
  tags                     = var.resource_tag
  account_tier             = var.account_tier
  account_replication_type = var.account_type
  account_kind             = var.account_kind
}

resource "azurerm_storage_container" "storage_container" {
  count                 = length(var.storage_container_name)
  name                  = var.storage_container_name[count.index]
  storage_account_name  = azurerm_storage_account.storage_account.name
  container_access_type = var.container_access
}

container_registry:
resource "azurerm_container_registry" "registry" {
  name                = var.registry_name
  resource_group_name = var.rg_name
  location            = var.location
  sku                 = var.sku
  admin_enabled       = true
  tags                = var.resource_tag
}

```

### Debug Output
```
Error: Plugin did not respond
│
│   with module.container.azurerm_container_registry.registry,
│   on modules\container\main.tf line 2, in resource "azurerm_container_registry" "registry":
│    2: resource "azurerm_container_registry" "registry" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.front_door.azurerm_frontdoor.front_door,
│   on modules\load_balancer\main.tf line 2, in resource "azurerm_frontdoor" "front_door":
│    2: resource "azurerm_frontdoor" "front_door" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.storage_account.azurerm_storage_account.storage_account,
│   on modules\storage\main.tf line 2, in resource "azurerm_storage_account" "storage_account":
│    2: resource "azurerm_storage_account" "storage_account" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v2.75.0_x5.exe plugin:

panic: interface conversion: interface {} is []interface {}, not string

goroutine 59 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenRoutingRuleForwardingConfiguration(0x6865e08, 0xc0017c8240, 0x5715ae0, 0xc001d94108, 0x4, 0xc001b20bb0, 0xad)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:2005 +0xe7f
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenSingleFrontDoorRoutingRule(0xc0018883c0, 0xc002296bc0, 0xc002296bd0, 0xc002296bb0, 0x5715ae0, 0xc001d94108, 0xc0017205af, 0x24, 0xc0017205e3, 0x10, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:1916 +0x7ad
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.flattenFrontDoorRoutingRule(0xc001d945d0, 0x5715ae0, 0xc001d94108, 0xc0017205af, 0x24, 0xc0017205e3, 0x10, 0xc00172061b, 0x12, 0x9f98a40, ...)    
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:1884 +0x2e5
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.resourceFrontDoorRead(0xc0012d0480, 0x58c2380, 0xc001282e00, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_resource.go:717 +0x131e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0009cac40, 0x6898ee8, 0xc0014ccbc0, 0xc0012d0480, 0x58c2380, 0xc001282e00, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:335 +0x1fb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0009cac40, 0x6898ee8, 0xc0014ccbc0, 0xc0006c0150, 0x58c2380, 0xc001282e00, 0xc0001d6ad0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:624 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000004f18, 0x6898ee8, 0xc0014ccbc0, 0xc0014ccc00, 0xc0014ccbc0, 0x2abeca, 0x5bf18e0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:575 +0x45b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc001b69440, 0x6898f90, 0xc0014ccbc0, 0xc001408420, 0xc001b69440, 0xc001750c00, 0xc000b93ba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:298 +0x108
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x5db5900, 0xc001b69440, 0x6898f90, 0xc001750c00, 0xc0014083c0, 0x0, 0x6898f90, 0xc001750c00, 0xc00176c000, 0xec8)       
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x222  
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00027ae00, 0x68e81f8, 0xc000914180, 0xc00172d0e0, 0xc001273950, 0x9f421f0, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00027ae00, 0x68e81f8, 0xc000914180, 0xc00172d0e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00090ea10, 0xc00027ae00, 0x68e81f8, 0xc000914180, 0xc00172d0e0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.75.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Expected Behaviour
 plan should be created

### Actual Behaviour
Crash in the plugin.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform init`
2. `terraform plan`

--- a/internal/services/frontdoor/frontdoor_resource.go
+++ b/internal/services/frontdoor/frontdoor_resource.go
@@ -1968,7 +1968,7 @@ func flattenRoutingRuleForwardingConfiguration(config frontdoor.BasicRouteConfig
 	cacheQueryParameterStripDirective := string(frontdoor.StripAll)
 	cacheUseDynamicCompression := false
 
-	var cacheQueryParameters *string
+	var cacheQueryParameters []interface{}
 	var cacheQueryParametersArray []string
 	var cacheDuration *string
 
@@ -2002,8 +2002,10 @@ func flattenRoutingRuleForwardingConfiguration(config frontdoor.BasicRouteConfig
 							cacheUseDynamicCompression = ofc["cache_use_dynamic_compression"].(bool)
 							cacheDuration = utils.String(ofc["cache_duration"].(string))
 
-							cacheQueryParameters = utils.String(ofc["cache_query_parameters"].(string))
-							cacheQueryParametersArray = strings.Split(*cacheQueryParameters, ",")
+							cacheQueryParameters = ofc["cache_query_parameters"].([]interface{})
+							for _, p := range cacheQueryParameters {
+								cacheQueryParametersArray = append(cacheQueryParametersArray, p.(string))
+							}
 						}
 					}
 				}

